### PR TITLE
WIP - Fix to ensure add element button works

### PR DESCRIPTION
**Pull request by stesi561**
Overview
----------------------------------------
TODO

Before
----------------------------------------
Clicking on the Add element doesn't work properly in some instances.
image

After
----------------------------------------
Clicking on the Add element works as expect.

Technical Details
----------------------------------------
TODO
----------------------------------------

@jitendrapurohit has made this fix however is currently away. Posting here in case someone else also finds this issue in the interim.

### CODE REVIEW

File: includes/utils.inc [hunk 1]
```diff
@@ -506,6 +506,7 @@ function wf_crm_array2str($arr) {
  *   Result of API call
  */
 function wf_civicrm_api($entity, $operation, $params) {
+  \Drupal::service('civicrm')->initialize();
```
Comment (mglaman) [inline includes/utils.inc:509]: This needs to go into `CivicrmOptions::prepare` which is being invoked for some reason during the add element process.

File: src/Plugin/WebformElement/CivicrmOptions.php [hunk 2]
```diff
@@ -9,6 +9,8 @@
 use Drupal\webform\WebformSubmissionInterface;
 use Drupal\webform_civicrm\Utils;
 
+include_once __DIR__ . '/../../../includes/utils.inc';
```
Comment (mglaman) [inline src/Plugin/WebformElement/CivicrmOptions.php:12]: Valid! Per the get field options method which invokes the helper function.

**Review by mglaman — CHANGES_REQUESTED** [review]
We can prevent a lot of initialization calls and checks if we adjust the invocation to when it is needed

File: includes/utils.inc (hunk 1 above)
Comment (stesi561) [reply includes/utils.inc:509]: Do you want me to update and test this or is this something you have already done?

Comment (KarinG) [reply includes/utils.inc:509]: I’ll do that - and test it -I was able to reproduce this issue

Comment (mglaman) [reply includes/utils.inc:509]: :) thanks @KarinG. Only the author and maintainers can push to PRs

Comment (jitendrapurohit) [reply includes/utils.inc:509]: Is it already modified and pushed as a PR? would be happy to do a quick test if it is needed :)

Comment (KarinG) [reply includes/utils.inc:509]: @jitendrapurohit - sorry - no I don’t think I got to it before holidays and still catching up. If you can make the edit - I promise it’s top of the list for me to test / merge.

Comment (jitendrapurohit) [reply includes/utils.inc:509]: I've submitted a new one - https://github.com/colemanw/webform_civicrm/pull/242

**Comment by KarinG** [comment]
Closing in favour of: https://github.com/colemanw/webform_civicrm/pull/242

**Pull request closed without merging**